### PR TITLE
NNS1-2965 move source code to navigation

### PR DESCRIPTION
**Pull request by coskucinkilic**
# Motivation

Make SouceCodeButton more visible, also change SourceCodeButton to match with new Figma files. Prepare the MenuItems to upcoming changes for themeToggle moving from account menu to sidebar as well and TVL display in sidebar

# Changes

SourceCodeButton moved from account menu to sidebar
SourceCodeButton moved from header folder to common folder 
Footer div is added to MenuItems to accomodate upcoming changes
MenuMetrics styling has been updated as paddings moved to menu footer in MenuItems.
Screenshot 2024-09-16 at 17 05 58
Screenshot 2024-09-16 at 17 06 09
with TVL:
Screenshot 2024-09-17 at 10 33 57
Screenshot 2024-09-17 at 10 33 50

# Tests

Tests related to SourceCodeButton has been moved from AccountMenu to MenuItems.

# Todos

- [x] Add entry to changelog (if necessary).

### CODE REVIEW

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 1]
```diff
@@ -94,10 +95,41 @@
...
+  </div>
 </TestIdWrapper>
+
+<style lang="scss">
+  .menu-container {
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    height: 100%;
+    margin-bottom: var(--padding-8x);
```
Comment (mstrasinskis) [inline frontend/src/lib/components/common/MenuItems.svelte:120]: Could you please add a comment why this is necessary?

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 2]
```diff
@@ -94,10 +95,41 @@
...
+  .menu-footer {
+    display: flex;
+    flex-direction: column;
+    justify-self: flex-end;
+  }
+
+  .menu-footer-buttons {
+    display: grid;
+    grid-template-columns: 3fr 1fr;
+    gap: 8px;
```
Comment (mstrasinskis) [inline frontend/src/lib/components/common/MenuItems.svelte:132]: Please use --padding

File: frontend/src/lib/components/common/SourceCodeButton.svelte [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<script lang="ts">
+  import { i18n } from "$lib/stores/i18n";
+  import { IconGitHub } from "@dfinity/gix-components";
+  function colorToRgb(color: string): string {
```
Comment (dskloetd) [inline frontend/src/lib/components/common/SourceCodeButton.svelte:4]: Why do we need this?

**Review by dskloetd — COMMENTED** [review]
The PR looks a lot more complicated than the PR description suggests. Can you provide some more context in the PR description?
Also fill out the "Tests" section please. You can mention unit tests or manual testing or put screenshots. As long as it's clear that it's sufficiently tested.

And shouldn't we move the theme switcher as well? Does the footer look OK with just the source code button moved? The repo should always be in a state where we can release.

If just moving the source code button puts the repo in a state where we can't release, but moving both is too much change for 1 PR, we have several options:
1. Use a feature flag to hide the changes until all the PRs are done.
2. Make separate PRs which add the new buttons to the code base but don't use them yet. Then make one final PR which uses the new buttons.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 1 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte:120]: You mean here or in the code?

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 2 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte]: fixed my mistake

File: frontend/src/lib/components/common/SourceCodeButton.svelte (hunk 3 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/SourceCodeButton.svelte]: definitely we don't need this, my bad it was one my trials. fixed now

**Comment by coskucinkilic** [comment]
@dskloetd After talking with Max, and he raising similar concerns I made sure that this PR can be merged on it's own and looks decent on the webpage.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 1 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte]: I have changed the code to make it simpler and still achieve the same styling please take another look.

File: CHANGELOG-Nns-Dapp-unreleased.md
Comment (dskloetd) [inline CHANGELOG-Nns-Dapp-unreleased.md:1]: Please don't reformat the entire file.

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 4]
```diff
@@ -87,17 +88,44 @@
...
 
 <TestIdWrapper testId="menu-items-component">
-  {#each routes as { context, title, href, icon, statusIcon, selected } (context)}
-    <MenuItem {href} testId={`menuitem-${context}`} {selected} {title}>
-      <svelte:component this={icon} slot="icon" />
-      <svelte:fragment>{title}</svelte:fragment>
-      <svelte:component this={statusIcon} slot="statusIcon" />
-    </MenuItem>
-  {/each}
+  <div class="menu-container">
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:91]: You can merge `<TestIdWrapper>` and the `<div>` below it into a single `<div>`.
```
<div data-tid="menu-items-component" class="menu-container">
```

I added the `TestIdWrapper` component when I added the page objects to have a quick way to add `data-tid` attributes to components with multiple top-level elements without worrying about the style by adding `display: contents`.
If the component has a single top-level element, you can simply add a `data-tid` attribute to that.

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 5]
```diff
@@ -87,17 +88,44 @@
...
+    {#each routes as { context, title, href, icon, statusIcon, selected } (context)}
+      <MenuItem {href} testId={`menuitem-${context}`} {selected} {title}>
+        <svelte:component this={icon} slot="icon" />
+        <svelte:fragment>{title}</svelte:fragment>
+        <svelte:component this={statusIcon} slot="statusIcon" />
+      </MenuItem>
+    {/each}
+    {#if IS_TESTNET}
+      <GetTokens />
+    {/if}
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:97]: Did you remove this intentionally?

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 6]
```diff
@@ -87,17 +88,44 @@
...
+
+  .menu-footer {
+    margin-top: auto;
+    flex-direction: column;
+    justify-self: flex-end;
+    margin-bottom: var(--padding-8x);
+  }
+
+  .menu-footer-buttons {
+    display: grid;
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:127]: Why grid?

**Review by dskloetd — COMMENTED** [review]
Can you add a screenshot of the area to the PR description?
I'd like to see what it looks like with just the source code button and not the theme switcher before I approve.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 6 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte:127]: There was fragments 3 to 1 ratio, I deleted them forgot the grid.I am deleting grid and the gap, will add them later on once I move the toggle as well.

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 7]
```diff
@@ -86,7 +87,7 @@
   ];
 </script>
 
-<TestIdWrapper testId="menu-items-component">
+<div data-tid="menu-items-component" class="menu-container">
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:89]: Also remove the `import TestIdWrapper`.

**Review by dskloetd — COMMENTED** [review]
The code looks good, but this is still unaddressed:

> Can you add a screenshot of the area to the PR description?
I'd like to see what it looks like with just the source code button and not the theme switcher before I approve.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 1 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/common/MenuItems.svelte]: I mean before `margin-bottom: var(--padding-8x);` to know why do we need this big gap here. I believe it’s due to the `100% on-chain` logo, and you showed me the calculations. It would be helpful to present them here so that if we update the SVG, we can follow the same logic to adjust this value.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 2 above)
Comment (mstrasinskis) [reply frontend/src/lib/components/common/MenuItems.svelte]: Thanks

**Comment by coskucinkilic** [comment]
Incase you want to have a look at the devenv deployment here is the link -> https://qsgjb-riaaa-aaaaa-aaaga-cai.cosku-ingress.devenv.dfinity.network

**Comment by dskloetd** [comment]
> Incase you want to have a look at the devenv deployment here is the link -> https://qsgjb-riaaa-aaaaa-aaaga-cai.cosku-ingress.devenv.dfinity.network/

Link currently gives me a 502.

Thanks for adding the screenshot to the PR description. Have you also checked how it looks with the "Total value locked" enabled? Would be good to make sure it looks right.

**Comment by coskucinkilic** [comment]
I stopped the devenv by mistake, will be running it again now. Let me upload another screenshot with TVL

**Comment by dskloetd** [comment]
Thanks. I the UI mocks the button and TVL widget have the same background color. But in the screenshot they look off. Should they not be the same?

**Comment by coskucinkilic** [comment]
They should be the same, but this is not the ticket related to TVL, I just adjust the paddings to make sure that it doesn't look strange. I believe there is a already open TVL PR, maybe we can include the colors there.

**Comment by dskloetd** [comment]
Yes, just the color. Thanks!

**Comment by coskucinkilic** [comment]
Colors have been updated now, TVL colors match with Figma.

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 8]
```diff
@@ -99,5 +100,39 @@
...
+    height: 100%;
+  }
+
+  .menu-footer {
+    display: none;
+    flex-direction: column;
+    gap: var(--padding);
+    // To accomodate the 100% on-chain logo
+    // if that logo changes please update this margin as well
+    margin: auto var(--padding-3x) var(--padding-8x) 0;
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:123]: Why is this necessary? Does the logo not take up space itself?

File: frontend/src/lib/components/common/MenuItems.svelte [hunk 9]
```diff
@@ -99,5 +100,39 @@
...
+    // Handle menu collapse animation
```
Comment (dskloetd) [inline frontend/src/lib/components/common/MenuItems.svelte:124]: How does the TVL widget currently handle the menu collapse animation? Is this copied from somewhere else? Can it be removed somewhere else?

File: frontend/src/lib/components/metrics/TotalValueLocked.svelte [hunk 10]
```diff
@@ -59,37 +58,26 @@
...
-    background: rgba(var(--background-disable-rgb), 0.1);
-    color: var(--body-color);
-
+    background: var(--sidebar-button-background);
     border-radius: var(--border-radius);
 
     width: 100%;
     padding: var(--padding) var(--padding-2x);
 
+    &:hover {
```
Comment (dskloetd) [inline frontend/src/lib/components/metrics/TotalValueLocked.svelte:69]: The TVL widget is not a button. Should it really change color on hover?

Comment (coskucinkilic) [reply frontend/src/lib/components/metrics/TotalValueLocked.svelte:69]: Aaccording to design it does

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 9 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte:124]: TVL widget still has it's own animation. which this one is derived from. we can move all the animations to MenuItems in the end.

File: frontend/src/lib/components/common/MenuItems.svelte (hunk 8 above)
Comment (coskucinkilic) [reply frontend/src/lib/components/common/MenuItems.svelte:123]: it doesn't

**Review by dskloetd — APPROVED** [review]
Approving to unblock, but let's keep discussing these last points.
